### PR TITLE
Add guide for Docker, Traefik & Letsencrypt

### PR DESCRIPTION
**Pull request by mvdstam**
As requested by @Idez in #1922. I hope to contribute more in the future regarding documenting Traefik. 👍

### CODE REVIEW

**Comment by ldez** [comment]
@mvdstam Thanks!

Could fix the build?

> Errors from misspell:
> docs/user-guide/getting-started-with-docker-and-lets-encrypt.md:86:12: "mimimum" is a misspelling of "minimum"
>
> Please fix the above errors. You can test via "misspell" and commit the result.
```

**Comment by mvdstam** [comment]
@ldez Done!

File: docs/user-guide/getting-started-with-docker-and-lets-encrypt.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
+# Docker & Traefik
+In this use case, we want to use Traefik as a *layer-7* load balancer with SSL termination for a set of microservices used to run a webapplication. We also want to automatically *discover any services* on the Docker host and let Traefik reconfigure itself automatically when containers get created (or shut down) so HTTP traffic can be routed accordingly. In addition, we want to use Let's Encrypt to automatically generate and renew SSL certificates per hostname.
+
+## Setting up
+In order for this to work, you'll need a server with a public IP address, with Docker installed on it. In this example, we're using the fictitious domain *my-awesome-app.org". In real-life, you'll want to use your own domain and have the DNS configured accordingly so the hostname records you'll want to use point to the aforementioned public IP address.
```
Comment (nmengin) [inline docs/user-guide/getting-started-with-docker-and-lets-encrypt.md]: The character **"** has to be replaced by a character _*_ to terminate the *italic* section.

File: docs/user-guide/getting-started-with-docker-and-lets-encrypt.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+Now, let's create a directory on the server where we will configure the rest of Traefik:
+
+`$ mkdir -p /opt/traefik`
+
+Within this directory, we're going to create 3 empty files:
+
+```sh
+$ touch /opt/traefik/docker-compose.yml
+$ touch /opt/traefik/acme.json
```
Comment (nmengin) [inline docs/user-guide/getting-started-with-docker-and-lets-encrypt.md]: The acme.json file must only be accessed in read/wrie by the owner that why it should be better to set these commands : 
`touch /opt/traefik/acme.json  && chmod 600 /opt/traefik/acme.json`

File: docs/user-guide/getting-started-with-docker-and-lets-encrypt.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+The `docker-compose.yml` file will provide us with a simple, consistent and more importantly, a deterministic way to create Traefik. The contents of the file is as follows:
+
+```yaml
+version: '2'
+
+services:
+  traefik:
+    image: traefik:1.3.5
+    restart: always
```
Comment (nmengin) [inline docs/user-guide/getting-started-with-docker-and-lets-encrypt.md]: The `always` policy may be replaced by the `unless-stopped` or `on-failure` policies.

File: docs/user-guide/getting-started-with-docker-and-lets-encrypt.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+entryPoint = "https"
+OnHostRule = true
+```
+
+This is the minimum configuration required to do the following:
+
+- Log `ERROR`-level messages (or more severe) to the console, but silence `DEBUG`-level messagse
+- Check for new versions of Traefik periodically
+- Create two entrypoints, namely an `HTTP` endpoint on port `80`, and an `HTTPS` endpoint on port `443` where all incoming traffic on port `80` will immediately get redirected to `HTTPS`.
+- Enable the Docker configuration backend and listen for container events on the Docker unix socket we've mounted earlier. However, **new containers will not be exposed by Traefik by default, we'll get into this in a bit!*
```
Comment (nmengin) [inline docs/user-guide/getting-started-with-docker-and-lets-encrypt.md]: A _*_ character is missing at the end of the line to terminate the **bold** section.

**Comment by mvdstam** [comment]
@nmengin I've updated the PR with your requested changes. The remark you made regarding the `restart` policy seems unnecessary as this is a well-documented (and well-understood) feature within Docker and I see no reason to change that in the example I described in the document. Especially since the example describes the setup of a front-facing reverse proxy like Traefik, I wouldn't understand why you wouldn't want the restart policy on anything else than `always`.

**Comment by nmengin** [comment]
@mvdstam It was just a suggestion. Of course, for a user guide, the type of restart policy is not really important. I should say `might be` instead of `may be` ;).

**Comment by ldez** [comment]
@mvdstam could you authorize us to modify your PR (rebase)?
https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/

**Comment by mvdstam** [comment]
@ldez Done!

**Pull request merged**